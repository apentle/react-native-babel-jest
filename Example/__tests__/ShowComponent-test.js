'use strict';

import React, { View, Text } from 'react-native';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

jest.unmock('../ShowComponent');
const ShowComponent = require('../ShowComponent');

describe('<ShowComponent />', () => {
  it('Component must show "Show Text"', () => {
    // Shallow rendering
    const output = shallow(<ShowComponent />);
    expect(output.type()).toBe(View);
    expect(output.find(Text).length).toBe(1);

    // Snapshot testing
    expect(shallowToJson(output)).toMatchSnapshot();

    // Full DOM rendering
    const wrapper= mount(<ShowComponent />);
    expect(wrapper.find(Text).text()).toBe('Show Text');
  });
});
