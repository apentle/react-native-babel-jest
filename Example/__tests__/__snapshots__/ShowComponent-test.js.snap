exports[`<ShowComponent /> Component must show "Show Text" 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }>
    Show Text
  </Text>
</View>
`;
